packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FGMGroundOverlayController.m

```diff
@@ -0,0 +1,240 @@
+                          coordinate:CLLocationCoordinate2DMake(
+                                         groundOverlay.bounds.southwest.latitude,
+                                         groundOverlay.bounds.southwest.longitude)]
+                             icon:[FGMUtils iconFromBitmap:groundOverlay.image
+                                                 registrar:self.registrar
+                                               screenScale:[self getScreenScale]]];
+    } else {
+      NSAssert(groundOverlay.zoomLevel != nil,
+               @"If ground overlay is initialized with position, zoomLevel is required");
+      gmsOverlay = [GMSGroundOverlay
+          groundOverlayWithPosition:CLLocationCoordinate2DMake(groundOverlay.position.latitude,
+                                                               groundOverlay.position.longitude)
+                               icon:[FGMUtils iconFromBitmap:groundOverlay.image
+                                                   registrar:self.registrar
+                                                 screenScale:[self getScreenScale]]
+                          zoomLevel:[groundOverlay.zoomLevel doubleValue]];
+    }
+    FGMGroundOverlayController *controller =
+        [[FGMGroundOverlayController alloc] initWithGroundOverlay:gmsOverlay
+                                                       identifier:identifier
+                                                          mapView:self.mapView
+                                              isCreatedWithBounds:isCreatedWithBounds];
+    controller.zoomLevel = groundOverlay.zoomLevel;
+    [controller updateFromPlatformGroundOverlay:groundOverlay
+                                      registrar:self.registrar
+                                    screenScale:[self getScreenScale]];
+    self.groundOverlayIdentifierToController[identifier] = controller;
+  }
+}
+
+- (void)changeGroundOverlays:(NSArray<FGMPlatformGroundOverlay *> *)groundOverlaysToChange {
+  for (FGMPlatformGroundOverlay *groundOverlay in groundOverlaysToChange) {
+    NSString *identifier = groundOverlay.groundOverlayId;
+    FGMGroundOverlayController *controller = self.groundOverlayIdentifierToController[identifier];
+    [controller updateFromPlatformGroundOverlay:groundOverlay
+                                      registrar:self.registrar
+                                    screenScale:[self getScreenScale]];
+  }
+}
+
+- (void)removeGroundOverlaysWithIdentifiers:(NSArray<NSString *> *)identifiers {
+  for (NSString *identifier in identifiers) {
+    FGMGroundOverlayController *controller = self.groundOverlayIdentifierToController[identifier];
+    if (!controller) {
+      continue;
+    }
+    [controller removeGroundOverlay];
+    [self.groundOverlayIdentifierToController removeObjectForKey:identifier];
+  }
+}
+
+- (void)didTapGroundOverlayWithIdentifier:(NSString *)identifier {
+  if (!identifier) {
```

Comment:
Here and in the next method: the identifier params were annotated as non-nullable in the interface, so the implementation code shouldn't be handling nil.

Reply:
Good catch. Removed the null check.

Reply:
It's still here in this method.